internal/apps/setup-smoke-test/Dockerfile

```diff
@@ -42,6 +42,12 @@ RUN set -ex; if [ "$build_env" = "alpine" ] && [ "$build_with_cgo" = "1" ]; then
       apk update && apk add gcc libc-dev; \
     fi
 
+# If requested, upgrade go-libddwaf to the desired release.
+ARG go_libddwaf_ref=""
+RUN if [ "${go_libddwaf_ref}" != "" ]; then \
+    go get -u github.com/DataDog/go-libddwaf/v2@${go_libddwaf_ref}; \
+  fi
```

Comment:
```suggestion
    go mod tidy
  fi
```

Reply:
There is a `go mod tidy` done literally just after this unconditionally...